README.md

```diff
@@ -37,6 +37,236 @@ flutterfire --help
+--ios-build-config=Debug \
+--ios-out=/ios/debug/GoogleService-Info.plist \
+--ios-bundle-id=com.example.myApp.ios.debug \
+--macos-build-config=Debug \
+--macos-out=/macos/debug/GoogleService-Info.plist \
+--macos-bundle-id=com.example.myApp.macos.debug \
+--web-app-id=<1:XXX:web:XXX> \
+--out=lib/debug/firebase_options.dart
+```
+
+Let's create a release build type for your Flutter app, here's how it might look:
+
+```bash
+flutterfire configure \ 
+--yes \
+--project=your-project-id \
+--platforms=ios,macos,android,web \
+--android-out=android/app/src/release/google-services.json \
+--android-package-name=com.example.myapp.release \
+--ios-build-config=Release \
+--ios-out=/ios/release/GoogleService-Info.plist \
+--ios-bundle-id=com.example.myApp.ios.release \
+--macos-build-config=Release \
+--macos-out=/macos/release/GoogleService-Info.plist \
+--macos-bundle-id=com.example.myApp.macos.release \
+--web-app-id=<1:XXX:web:XXX> \
+--out=lib/release/firebase_options.dart
+```
+
+**Important things to note with multi-environment set up** 
+
+#### Android
+- The android gradle plugin will try to match the path `google-services.json` to the correct build type. For instance, debug build will find the service file in the following locations: `android/app/src/debug/google-services.json` or `android/app/google-services.json`.
+- Required to configure the application ID for release build in your android `build.gradle`. This might look like the following:
+
+```gradle
+buildTypes {
+    release {
+        signingConfig signingConfigs.debug
+        // This will effectively add the suffix to your applicationId. e.g. "com.example.myapp.release"
+        applicationIdSuffix ".release"
+    }
+}
+```
+
+#### Apple
+- Required to configure the bundle ID for release build configuration in Xcode for macOS & iOS under "Signings and Capabilities" tab. Below is an image of where you can update the release build configuration bundle ID:
+![Xcode release build configuration bundle ID](images/xcode-build-configuration.png)
+
+#### Web
+- Required to manually create a Firebase web app for release and use the Firebase web App ID as an argument for `--web-app-id`.
+- Required to implement your own logic for configuring build types for your Flutter web app as mentioned above.
+
+If you require further granular builds i.e. build flavors, you can refer to [this article](https://codewithandrea.com/articles/flutter-flavors-for-firebase-apps/) which demonstrates how to configure different build flavors using [VGV CLI](https://github.com/VeryGoodOpenSource/very_good_cli) to create a Flutter app that has build flavors out of the box. 
+
+### Flutterfire reconfigure
+
+There is another command that you might want to run each time you configure your project with another Firebase product. For example, if you configure Real time database for your Firebase project, you will be required to reinstall every service file for each platform. By running:
```

Comment:
I think it's also important that you need to run `flutterfire reconfigure` it you add later `firebase_performance` or `firebase_crashlytics` to your project. This is something that is super easy to oversee because you don't get any errors/warning when you run:

```sh
# First FlutterFire configure
flutterfire configure

# And then add firebase_crashlytics
flutter pub add firebase_crashlytics
```

In this case, the 'FlutterFire: "flutterfire upload-crashlytics-symbols"' build phase in XCode is missing.

Something for the future: Maybe a better user experience would be to add another build phase that checks, if the project uses `firebase_crashlytics` and if yes -> check if 'FlutterFire: "flutterfire upload-crashlytics-symbols"' is present in XCode, if no -> add it automatically or print a warning.

Reply:
This is also something that needs to be mentioned in the Firebase Analytics docs

Reply:
Thanks for the feedback, @nilsreichardt! I've made some updates based on your comments 👍